server/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java

```diff
@@ -527,8 +531,9 @@ public void apply(Settings value, Settings current, Settings previous) {
         ReadinessService.PORT,
         HealthNodeTaskExecutor.ENABLED_SETTING,
         LocalHealthMonitor.POLL_INTERVAL_SETTING,
-        TransportHealthNodeAction.HEALTH_NODE_TRANSPORT_ACTION_TIMEOUT
-    );
+        TransportHealthNodeAction.HEALTH_NODE_TRANSPORT_ACTION_TIMEOUT,
+        TcpTransport.isUntrustedRemoteClusterEnabled() ? SniffConnectionStrategy.REMOTE_CLUSTER_AUTHORIZATION : null
```

Comment:
Is this not going to NPE without the feature flag enable?

Reply:
These lines of code are copied from a previous Health API PR to allow adding nulls, and then filter them out before collect.

https://github.com/elastic/elasticsearch/blob/437a456dc17c48ed0a30b61d33e1baeee5e8161a/server/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java#L192
https://github.com/elastic/elasticsearch/blob/437a456dc17c48ed0a30b61d33e1baeee5e8161a/server/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java#L536